src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -14,35 +14,51 @@
             [utils.number]))
 
 (rf/reg-event-fx :wallet.swap/start
- (fn [{:keys [db]} [{:keys [asset-to-pay asset-to-receive network]}]]
-   (let [asset-to-receive' (or asset-to-receive
-                               (swap-utils/select-asset-to-receive
-                                (:wallet db)
-                                (:profile/profile db)
-                                asset-to-pay))
-         network'          (or network
-                               (swap-utils/select-network asset-to-pay))]
+ (fn [{:keys [db]} [{:keys [asset-to-pay asset-to-receive network navigate-within-stack?]}]]
+   (let [{:keys [wallet]}       db
+         test-networks-enabled? (get-in db [:profile/profile :test-networks-enabled?])
+         account                (swap-utils/wallet-account wallet)
+         asset-to-pay'          (if (:networks asset-to-pay)
+                                  asset-to-pay
+                                  (swap-utils/select-asset-to-pay-by-symbol
+                                   {:wallet                 wallet
+                                    :account                account
+                                    :test-networks-enabled? test-networks-enabled?
+                                    :token-symbol           (:symbol asset-to-pay)}))
+         asset-to-receive'      (or asset-to-receive
+                                    (swap-utils/select-default-asset-to-receive
+                                     {:wallet                 wallet
+                                      :account                account
+                                      :test-networks-enabled? test-networks-enabled?
+                                      :asset-to-pay           asset-to-pay'}))
+         network'               (or network
+                                    (swap-utils/select-network asset-to-pay'))]
```

Comment:
to avoid shadowing and get rid of `'` in the names, would be cleaner to not destruct these keys in the args.

Reply:
Yeah @clauxx, also they are completely out of the Clojure naming covention.

CC: @ilmotta